File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.go
func dividePlayers(skill []int) int64 {
    
    chemistry := 0
    slices.Sort(skill)
    start, end := 0, len(skill) - 1
    teamSum := skill[start] + skill[end]
    for start < end {
        if teamSum != skill[start] + skill[end]{
            return -1
        }
        chemistry += (skill[start] * skill[end])
        start += 1
        end -= 1
    }
    
    return int64(chemistry)
}

File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        n = len(skill)
        start = chemistry = 0
        end = n - 1
        curr_sum = skill[0] + skill[-1]
        while(start < end):
            if curr_sum != (skill[start] + skill[end]):
                return -1
            
            chemistry += (skill[start] * skill[end])
            start += 1
            end -= 1
        
        return chemistry
            
            